### PR TITLE
refactor: Mod information

### PR DESCRIPTION
**Pull request by Alystrasz**
Mod counterpart of https://github.com/R2Northstar/NorthstarLauncher/pull/826.

### CODE REVIEW

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 1]
```diff
@@ -1136,36 +1108,58 @@ void function ThreadedAuthAndConnectToServer( string password = "", bool modsCha
...
+				string coreModVersion = NSGetModVersions( modName )[0]
+				if ( !NSIsModEnabled( modName ) )
+				{
+					modsChanged = true
+					NSSetModEnabled( modName, true )
+					print(format("Enabled \"%s\" (v%s) to join server.", modName, coreModVersion))
+				}
+			}
+
+			else if ( NSIsModRequiredOnClient( mod.name ) && !NSIsModEnabled( mod.name ))
```
Comment (Alystrasz) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:1158]: Actually, if I remember correctly, server only exposes mods that are client-required through `file.lastSelectedServer.requiredMods`, so the `NSIsModRequiredOnClient( mod.name )` check might be unnecessary.

**Review by Alystrasz — COMMENTED** [review]
On vc yesterday, Blue told `NSGetModVersions` and `NSIsModRequiredOnClient` methods are now useless, as all mod information should be accessed through `NSGetModsInformation`.
While I agree about `NSIsModRequiredOnClient`, I still think `NSGetModVersions` is required, notably in the main menu (we don't want to fetch info from all mods to display Northstar current version).
@RoyalBlue1 wdyt?

**Comment by RoyalBlue1** [comment]
> I still think `NSGetModVersions` is required, notably in the main menu (we don't want to fetch info from all mods to display Northstar current version).

Maybe instead `NSGetModVersions` use a function that can get all info for a singular mod might be more usefull while still minimizing the Api between Native and squirrel

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut (hunk 1 above)
Comment (Alystrasz) [reply Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut]: Check was indeed unnecessary and was subsequently removed.

**Comment by Alystrasz** [comment]
> > I still think `NSGetModVersions` is required, notably in the main menu (we don't want to fetch info from all mods to display Northstar current version).
> 
> Maybe instead `NSGetModVersions` creating a function that can get all info for a singular mod might be more usefull while still minimizing the Api between Native and squirrel

`NSGetModInformation` was added in d0286a904923bf98307901b110bc768c37faf277; NS mod getters/setters were removed in the following commits (with `NSSetModEnabled` as the only exception).

**Review by RoyalBlue1 — APPROVED** [review]
LGTM changes all the uses of the singular functions to get mod info to the struct

**Pull request merged**